caikit_nlp/modules/token_classification/filtered_span_classification.py

```diff
@@ -111,53 +132,97 @@ def run(
         if threshold is None:
             threshold = self.default_threshold
         token_classification_results = []
-        # Split document into spans
-        span_list = self.tokenizer.run(text)
+        if self.classification_task == TextClassificationTask:
+            # Split document into spans
+            span_list = self.tokenizer.run(text).results
+            text_list = [span.text for span in span_list]
+        else:
+            # TokenClassificationTask classifiers would hold span info
+            # so we do not need to span split again
+            text_list = [text]
```

Comment:
Yeah this is a bit tricky since I'm not sure how to "reuse" this info, since any given text classification task implementation (e.g. sequence classification) can do its own tokenization, or not. If there was a rule-based text classification implementation for some reason there would not be 2xtokenization